### PR TITLE
Use URLs instead of hosts in public API

### PR DESCRIPTION
**Pull request by kovayur**
## Description
This PR changes the Public API for using `centralUIURL` and `centralDataURL` instead of `ui_host` and `data_host`
cc: @sachaudh

## Checklist (Definition of Done)

- [x] ~Unit and integration tests added~
- [x] ~Documentation added if necessary~
- [x] CI and all relevant tests are passing
- [x] ~Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.~

## Test manual
Rely on CI
```
# To run tests locally run:
make db/teardown db/setup db/migrate
make ocm/setup OCM_OFFLINE_TOKEN=<ocm-offline-token> OCM_ENV=development
make verify lint binary test test/integration
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -358,9 +358,9 @@ test/e2e: gotestsum
 	go run github.com/onsi/ginkgo/v2/ginkgo -r $(GINKGO_FLAGS) \
 		--randomize-suites \
 		--fail-on-pending --keep-going \
-		--cover --coverprofile=cover.profile \
+		--cover --coverprofile=$(PROJECT_PATH)/data/results/cover.profile \
```
Comment (johannes94) [inline Makefile:361]: I think something with this change is wrong the test script failed locally and in CI with:
```
testing: open /Users/johannes/go/src/github.com/acs-fleet-manager/e2e/Users/johannes/go/src/github.com/acs-fleet-manager/data/results/cover.profile: no such file or directory
ginkgo run failed
  could not finalize profiles:
  Unable to read coverage file /Users/johannes/go/src/github.com/acs-fleet-manager/e2e/Users/johannes/go/src/github.com/acs-fleet-manager/data/results/cover.profile:
  open /Users/johannes/go/src/github.com/acs-fleet-manager/e2e/Users/johannes/go/src/github.com/acs-fleet-manager/data/results/cover.profile: no such file or directory
exit status 1
make: *** [test/e2e] Error 1

```

**Review by johannes94 — CHANGES_REQUESTED** [review]

File: e2e/e2e_test.go [hunk 2]
```diff
@@ -348,13 +355,13 @@ var _ = Describe("Central", func() {
 			}).WithTimeout(waitTimeout).WithPolling(defaultPolling).Should(Equal(newCentralResources))
 		})
 
-		It("should transition central's state to ready", func() {
+		It("[Admin] should transition central's state to ready", func() {
```
[the remark this answers is not in the document]
Comment (kovayur) [reply e2e/e2e_test.go]: Reverted.

File: go.mod [hunk 3]
```diff
@@ -81,9 +81,11 @@ require (
 	github.com/go-openapi/jsonpointer v0.19.5 // indirect
 	github.com/go-openapi/jsonreference v0.20.0 // indirect
 	github.com/go-openapi/swag v0.21.1 // indirect
+	github.com/go-task/slim-sprig v0.0.0-20210107165309-348f09dbbbc0 // indirect
```
[the remark this answers is not in the document]
Comment (kovayur) [reply go.mod]: Fixed by @mtesseract in #339

File: Makefile (hunk 1 above)
Comment (kovayur) [reply Makefile]: Thanks for testing, changed to a relative path

File: dev/env/scripts/bootstrap.sh [hunk 4]
```diff
@@ -57,6 +57,9 @@ fi
 
 if [[ "$INSTALL_OPERATOR" == "true" ]]; then
     if [[ "$INSTALL_OLM" == "true" ]]; then
+        if ! command -v operator-sdk; then
```
Comment (mtesseract) [inline dev/env/scripts/bootstrap.sh:60]: nit: redirect stdout/stderr to `/dev/null`?

File: fleetshard/pkg/central/reconciler/reconciler.go [hunk 5]
```diff
@@ -299,8 +299,6 @@ func (r CentralReconciler) ensureCentralDeleted(ctx context.Context, central *v1
 		return false, err
 	}
 	globalDeleted = globalDeleted && nsDeleted
-
-	glog.Infof("All central resources were deleted: %s/%s", central.GetNamespace(), central.GetName())
```
Comment (mtesseract) [inline fleetshard/pkg/central/reconciler/reconciler.go:303]: Is this not true anymore?

**Review by mtesseract — APPROVED** [review]

**Review by johannes94 — APPROVED** [review]
LGTM

File: fleetshard/pkg/central/reconciler/reconciler.go (hunk 5 above)
Comment (kovayur) [reply fleetshard/pkg/central/reconciler/reconciler.go:303]: Yes, this is not correct and misleading

File: dev/env/scripts/bootstrap.sh (hunk 4 above)
Comment (kovayur) [reply dev/env/scripts/bootstrap.sh:60]: Will fix in a separate PR

**Pull request merged**